src/machine/flash.go

```diff
@@ -91,21 +95,22 @@ func (fl *FlashBuffer) Write(p []byte) (n int, err error) {
 	// NOTE probably will not work as expected if you try to write over page boundary
 	// of pages with different sizes.
 	pagesize := uintptr(fl.b.EraseBlockSize())
-	currentPageCount := (fl.current - fl.start + pagesize - 1) / pagesize
-	totalPagesNeeded := (fl.current - fl.start + uintptr(len(p)) + pagesize - 1) / pagesize
-	if currentPageCount == totalPagesNeeded {
-		// just write the data
-		n, err := fl.b.WriteAt(p, int64(fl.current))
-		if err != nil {
-			return 0, err
-		}
-		fl.current += uintptr(n)
-		return n, nil
+
+	// calculate currentPageBlock relative to fl.start, meaning that
+	// block 0 -> fl.start
+	// block 1 -> fl.start + pagesize
+	// block 2 -> fl.start + pagesize*2
+	// ...
+	currentPageBlock := (fl.start + fl.offset - FlashDataStart()) / pagesize
+	lastPageBlockNeeded := (fl.start + fl.offset + uintptr(len(p)) - FlashDataStart()) / pagesize
+
+	// so we create at least 1 page
+	if currentPageBlock == lastPageBlockNeeded {
```

Comment:
Is this the write logic?  If I do a zero-byte write, should the function erase the block, or just return as a no-op?

Is this compensating for the change above that no longer supports partial page writes?

Reply:
Yes, so I can remove this by restoring the partial page write as you suggest.

Reply:
Done.